core/metacling/src/TCling.cxx

```diff
@@ -6675,167 +6675,101 @@ void TCling::UpdateListsOnCommitted(const cling::Transaction &T) {
-   TListOfFunctions* functions = (TListOfFunctions*)cl->GetListOfMethods();
-   TListOfEnums* enums = (TListOfEnums*)cl->GetListOfEnums();
-   TListOfFunctionTemplates* functiontemplates = (TListOfFunctionTemplates*)cl->GetListOfFunctionTemplates();
-   for (DeclContext::decl_iterator RI = DC->decls_begin(), RE = DC->decls_end(); RI != RE; ++RI) {
-      if (isa<VarDecl>(*RI) || isa<FieldDecl>(*RI)) {
-         const clang::ValueDecl* VD = dyn_cast<ValueDecl>(*RI);
-         var = (TDataMember*)datamembers->FindObject(VD->getNameAsString().c_str());
-         if (var) {
-            // Unload the global by setting the DataMemberInfo_t to 0
-            datamembers->Unload(var);
-            var->Update(0);
-         }
-      } else if (const FunctionDecl* FD = dyn_cast<FunctionDecl>(*RI)) {
-         function = (TFunction*)functions->FindObject(FD->getNameAsString().c_str());
-         if (function) {
-            functions->Unload(function);
-            function->Update(0);
-         }
-      } else if (const EnumDecl* ED = dyn_cast<EnumDecl>(*RI)) {
-         e = (TEnum*)enums->FindObject(ED->getNameAsString().c_str());
-         if (e) {
-            TIter iEnumConst(e->GetConstants());
-            while (TEnumConstant* enumConst = (TEnumConstant*)iEnumConst()) {
-               // Since the enum is already created and valid that ensures us that
-               // we have the enum constants created as well.
-               enumConst = (TEnumConstant*)datamembers->FindObject(enumConst->GetName());
-               if (enumConst && enumConst->IsValid()) {
-                  datamembers->Unload(enumConst);
-                  enumConst->Update(0);
-               }
-            }
-            enums->Unload(e);
-            e->Update(0);
-         }
-      } else if (const FunctionTemplateDecl* FTD = dyn_cast<FunctionTemplateDecl>(*RI)) {
-         functiontemplate = (TFunctionTemplate*)functiontemplates->FindObject(FTD->getNameAsString().c_str());
-         if (functiontemplate) {
-            functiontemplates->Unload(functiontemplate);
-            functiontemplate->Update(0);
-         }
-      }
-   }
-}
-
-////////////////////////////////////////////////////////////////////////////////
-
+///\brief Invalidate stored TCling state for declarations included in transaction `T'.
+///
 void TCling::UpdateListsOnUnloaded(const cling::Transaction &T)
 {
    HandleNewTransaction(T);
 
-   // Unload the objects from the lists and update the objects' state.
-   TListOfFunctions* functions = (TListOfFunctions*)gROOT->GetListOfGlobalFunctions();
-   TListOfFunctionTemplates* functiontemplates = (TListOfFunctionTemplates*)gROOT->GetListOfFunctionTemplates();
-   TListOfEnums* enums = (TListOfEnums*)gROOT->GetListOfEnums();
-   TListOfDataMembers* globals = (TListOfDataMembers*)gROOT->GetListOfGlobals();
+   auto Lists = std::forward_as_tuple(*(TListOfDataMembers *)gROOT->GetListOfGlobals(),
```

Comment:
Could you `forward_as_tuple` pointers and not references? Tuples of references always make me nervous...

Reply:
Updated.